src/Enums/UserRole.tsx

```diff
@@ -9,12 +9,14 @@ export enum UserRole {
     ADMIN     = 'ADMIN'
 }
 
-export const unauthorizedRedirect = () => {
+export const unauthorizedRedirect = (doRedirect: boolean = true) => {
     // TODO: Generic redirect to handle clearing cookies.
     general.loginRedirectURL = `${window.location.pathname}${window.location.search}`;
     Cookies.remove(CookieEnum.USERTYPE);
-    Cookies.remove(CookieEnum.SESSION);    
-    window.location.assign('/');
+    Cookies.remove(CookieEnum.SESSION);
+    if (doRedirect) {
+        window.location.assign('/');        
+    }
```

Comment:
👍

Reply:
When do we want to remove the session without redirection to login?

Reply:
@PanopticaRising react router returns a redirect component which does not cause another page reload
The code that was here did a page redirect
Since I made the unauthorized logic used in multiple places you would get both happening in some cases and the page would flicker

This is not the final solution, I would prefer doing something with global state that returned the redirect a higher level